ejb3/src/main/java/org/jboss/as/ejb3/component/interceptors/LogDiagnosticContextRecoveryInterceptor.java

```diff
@@ -64,12 +64,14 @@ public static LogDiagnosticContextRecoveryInterceptor getInstance() {
     }
 
     public Object processInvocation(final InterceptorContext context) throws Exception {
-        for (String str : MDC.getMap().keySet()) {
-            MDC.remove(str);
+        if (MDC.getMap() != null) {
+            for (String str : MDC.getMap().keySet()) {
```

Comment:
This is on the critical perf request path, and looking at the impl of MDC.getMap() it's pretty expensive, involving a map copy. So the result from the getMap() call at L67 should be cached and reused in L68.